internal/dataplane/translator/translator.go

```diff
@@ -188,6 +199,15 @@ func (t *Translator) BuildKongConfig() KongConfigBuildingResult {
 		t.registerSuccessfullyTranslatedObject(result.Plugins[i].K8sParent)
 	}
 
+	// process custom entities
+	result.FillCustomEntities(t.logger, t.storer, t.failuresCollector, t.schemaServiceProvider(), t.workspace)
+	// Register successcully translated KCEs to set the status of these KCEs.
+	for _, collection := range result.CustomEntities {
+		for i := range collection.Entities {
+			t.registerSuccessfullyTranslatedObject(collection.Entities[i].K8sKongCustomEntity)
+		}
+	}
```

Comment:
This should be guarded by the feature gate, right?

Reply:
Added feature gates check here.